### DISCUSSION TITLE
Best way to backup ryot

### DISCUSSION THREAD

**Discussion by kaibagley** [IgnisDa/ryot#670] [Q&A] [answered]
Hey

When taking a backup of ryot data, what is the best way. I've seen that I could save the postgres database somewhere, or using the export to s3 function in the UI.

I'm not sure what the best way to do this is. For all my other services I backup by either rsyncing the local directory used as the container's data/configs volume, or use built-in backup solutions and copy those to the backup drive.

**Comment by IgnisDa** [comment, marked as answer]
Dumping the postgres instance is 100% the best way to do it. All Ryot data is stored there.

**Reply by kaibagley** [reply]
This is what I'm using to write the postgres db: 
`docker exec -i ryot-db pg_dump -U postgres postgres > "$TARGET_DIRECTORY/$BACKUP_FILENAME`

I believe this is fine, I've dropped the `postgres` DB and replaced it with the above created backup and it all seemed to work well.

Thanks!!!
